### PR TITLE
Patch pdfs tiffs borked

### PR DESCRIPTION
**Pull request by jonahkaye**
Ticket: #[1350](https://github.com/metriport/metriport-internal/issues/1350)

### Description

- patching string conversion issues resulting in messed up pdfs and tiffs. Now, we decode b64 from DocumentReference into bytes and dont touch it and then write it back into the file on s3. 
- file size computed without filtering for utf-8, since tiffs and pdfs raw bytes filled with non utf-8
- preserving the fileType determination logic since it seems like it works 

### Testing

_[Regular PRs:]_

- Local
  - [x] created identical patients in local and staging. uploaded xml, pdf, tiff to patient on staging, then did: mirth outbound -> mirthInbound PD, DQ, DRs and looked at the downloaded files in the local patients s3 bucket and all the files downloaded fine

### Release Plan

- :warning: Points to `master`
- [ ] Merge this
- [ ] IHE gateway coordination

### CODE REVIEW

File: packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA ITI-39 Processor/destinationConnector-XCA Endpoint-responseTransformer-step-1-Case 1 - Success or PartialSuccess.js [hunk 1]
```diff
@@ -50,10 +50,16 @@ if ('Success' == queryResponseCode.toString() || 'PartialSuccess' == queryRespon
...
-					var documentEncodedString = entry.*::Document.toString();
-					var decoded = FileUtil.decode(documentEncodedString);
+					var documentEncodedString = entry.*::Document;
+					var decoded = FileUtil.decode(documentEncodedString.toString());
+					// we use decodedAsString to detect the file type
 					decodedAsString = new Packages.java.lang.String(decoded);
+
+					// We use decoded bytes to decode the actual raw document. 
+					var decoder = java.util.Base64.getDecoder();
+    				var decodedBytes = decoder.decode(documentEncodedString);
```
Comment (leite08) [inline packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA ITI-39 Processor/destinationConnector-XCA Endpoint-responseTransformer-step-1-Case 1 - Success or PartialSuccess.js:62]: This is already declared on line 53

File: packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA ITI-39 Processor/destinationConnector-XCA Endpoint-responseTransformer-step-1-Case 1 - Success or PartialSuccess.js [hunk 2]
```diff
@@ -84,13 +90,13 @@ if ('Success' == queryResponseCode.toString() || 'PartialSuccess' == queryRespon
 							if (title) attachment.title = title;
 						}
 					}
-					const fileSize = decodedAsString.getBytes("UTF-8").length;
+					const fileSize = decodedAsString.getBytes().length;
```
Comment (leite08) [inline packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA ITI-39 Processor/destinationConnector-XCA Endpoint-responseTransformer-step-1-Case 1 - Success or PartialSuccess.js:93]: NABD, but could we use `decodedBytes`, and so maybe save some cycles?

File: packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA ITI-39 Processor/destinationConnector-XCA Endpoint-responseTransformer-step-1-Case 1 - Success or PartialSuccess.js (hunk 1 above)
Comment (leite08) [reply packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA ITI-39 Processor/destinationConnector-XCA Endpoint-responseTransformer-step-1-Case 1 - Success or PartialSuccess.js:62]: Also, we should check this as we do `decodedAsString` on line 66, right?

File: packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA ITI-39 Processor/destinationConnector-XCA Endpoint-responseTransformer-step-1-Case 1 - Success or PartialSuccess.js [hunk 3]
```diff
@@ -114,6 +120,7 @@ if ('Success' == queryResponseCode.toString() || 'PartialSuccess' == queryRespon
 		}
 
 		// TODO: Process and generate OperationOutcome
+		logger.info("contentList " + contentList.length);
```
Comment (leite08) [inline packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA ITI-39 Processor/destinationConnector-XCA Endpoint-responseTransformer-step-1-Case 1 - Success or PartialSuccess.js:123]: left over or are do you really want to keep it in prod?

**Review by leite08 — APPROVED** [review]
Approved w/ a single comment

File: packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA ITI-39 Processor/destinationConnector-XCA Endpoint-responseTransformer-step-1-Case 1 - Success or PartialSuccess.js (hunk 3 above)
Comment (jonahkaye) [reply packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA ITI-39 Processor/destinationConnector-XCA Endpoint-responseTransformer-step-1-Case 1 - Success or PartialSuccess.js:123]: leftover good catch

**Pull request merged**